### ISSUE TITLE
No services discovered

### ISSUE THREAD

**Issue by i-h8-github** [tenderlove/zeroconf#4]
Hello,

I have tested this on Ubuntu, Raspberry Pi OS, and macOS, and on neither platform does it successfully discover any services. On Ubuntu and Pi, it simply returns nil, whereas on macOS an error regarding an IP already being in use crashes the program.

```ruby
require 'zeroconf'
all_services = ZeroConf.find_services
p all_services
```

mDNS is confirmed working on all three systems.

**Comment by tenderlove** [comment]
I think this is probably fixed on the latest version (v1.1.0).  Please give it a try.  If it's still broken, please let me know!

**Comment by i-h8-github** [comment]
It is now discovering some services, but not all.
